lib/tenstorrent/bh_arc/noc_dma.c

```diff
@@ -102,8 +102,9 @@ static uint32_t get_expected_acks(uint32_t noc_cmd, uint64_t size)
 static inline bool is_behind(uint32_t current, uint32_t target)
 {
 	uint32_t diff = current - target;
+	uint32_t uint32_middle = UINT32_MAX / 2 + 1;
 
-	return diff > DIV_ROUND_UP(UINT32_MAX, 2);
+	return diff > uint32_middle;
```

Comment:
what about `CLAMP()`? There are other static inline functions in Zephyr that will return `true` if an overflow occurs.